botocore/utils.py

```diff
@@ -3315,3 +3317,29 @@ def _serialize_if_needed(self, value, iso=False):
                 return value.isoformat()
             return value.strftime('%Y-%m-%dT%H:%M:%S%Z')
         return value
+
+
+def validate_compression_config(config):
+    disable_request_compression = config.disable_request_compression
+    if (
+        disable_request_compression is not None
+        and not isinstance(disable_request_compression, bool)
+        and not (
+            isinstance(disable_request_compression, str)
+            and disable_request_compression.lower() in ('true', 'false')
+        )
+    ):
```

Comment:
Adding to my sequence of comments about using "conversion_func" in `configparser.py`. I don't think we have equivalent code for any other config value and think we should follow the existing pattern.